### PR TITLE
feat: implements instance lock

### PR DESCRIPTION
**Pull request by evertonfraga**
#### What does it do?
it prevents grid from being launched more than once by any form (shortcut, taskbar icon, direct from executable).

#### Any helpful background information?
https://github.com/electron/electron/blob/f6a29707b64bc2f7364f89096d187246bfc53765/docs/api/app.md#apprequestsingleinstancelock

#### Relevant screenshots? 

videos.

before: 
https://cdn.discordapp.com/attachments/483981458705350658/603277081840713741/windows-issue.mp4

after:
https://cdn.discordapp.com/attachments/483981458705350658/603277438772051968/windows-instance-lock.mp4

after II:
https://cdn.discordapp.com/attachments/483981458705350658/603281176605491241/Untitled.mov

### CODE REVIEW

File: nano.js [hunk 1]
```diff
@@ -37,89 +37,86 @@ const mb = menubar({
...
       }
+    ])
+    mb.tray.on('right-click', () => {
+      mb.tray.popUpContextMenu(contextMenu)
+    })
+  })
+}
+
+const gotTheLock = mb.app.requestSingleInstanceLock()
+if (!gotTheLock) {
```
Comment (marcgarreau) [inline nano.js]: I think this code block can use some additional documentation: a brief description and/or a link to documentation here.

Comment (PhilippLgh) [reply nano.js]: hard to see the changes because a lot of untouched code is recognized as diff.. was the formatting changed?

Comment (evertonfraga) [reply nano.js]: yeah, most of the code is now on a `init()` function.

**Comment by evertonfraga** [comment]
Added documentation as @marcgarreau asked.

**Pull request merged**